scio-test/src/test/scala/com/spotify/scio/coders/CoderTest.scala

```diff
@@ -211,10 +211,9 @@ final class CoderTest extends AnyFlatSpec with Matchers {
       cw.runWithImplicit
       throw new Throwable("Is expected to throw when passing implicit from outer class")
     } catch {
-      case e: NullPointerException =>
-        // In this case outer field is called "$cw" and it is hard to wrap it with proper exception
-        // so we allow it to fail with NullPointerException
-        e.getMessage should be(null)
+      case _: NullPointerException =>
```

Comment:
Interestingly in Java 17, this now throws an NPE with the message `Cannot throw exception because "null" is null`. cc @shnapz @RustedBones

Reply:
@clairemcginty will take a look

Reply:
sounds good!